app/src/main/java/org/akvo/flow/service/bootstrap/BootstrapProcessor.kt

```diff
@@ -85,15 +85,15 @@ class BootstrapProcessor @Inject constructor(
             // now read the survey XML back into memory to see if there is a version
             val surveyMetadata = fileProcessor.readBasicSurveyData(surveyFile)
 
-            //most instances use the same for server base and instance url now
-            val instanceCodeName: String = if (BuildConfig.SERVER_BASE == BuildConfig.INSTANCE_URL) {
-                BuildConfig.AWS_BUCKET
+            if (surveyMetadata.alias.isNotEmpty() ) {
+                if (!BuildConfig.INSTANCE_URL.contains(surveyMetadata.alias)) {
+                    return ProcessingResult.ProcessingErrorWrongDashboard
+                }
+            } else if (surveyMetadata.app.isNotEmpty()) {
+                if (!BuildConfig.AWS_BUCKET.contains(surveyMetadata.app)) {
```

Comment:
I am starting to wonder if its not better add a bucket attribute in the XML. That way even if we have the same bucket its explicit that one is the bucket and the other is the appId.

Reply:
I am wondering... would it be useful? because as soon as these changes are added to the backend, the new alias will be added for all newly published forms. The old forms will keep having app attribute only, and that is why we still keep this check. In this scenario, I don't see the value of bucket, we would just use alias.